Index: cloud-init-0.7.8/setup.py
===================================================================
--- cloud-init-0.7.8.orig/setup.py
+++ cloud-init-0.7.8/setup.py
@@ -179,7 +179,6 @@ else:
         (ETC + '/cloud/cloud.cfg.d', glob('config/cloud.cfg.d/*')),
         (ETC + '/cloud/templates', glob('templates/*')),
         (ETC + '/NetworkManager/dispatcher.d/', ['tools/hook-network-manager']),
-        (ETC + '/dhcp/dhclient-exit-hooks.d/', ['tools/hook-dhclient']),
         (USR_LIB_EXEC + '/cloud-init', ['tools/uncloud-init',
                                         'tools/write-ssh-key-fingerprints']),
         (USR + '/share/doc/cloud-init', [f for f in glob('doc/*') if is_f(f)]),
@@ -211,7 +210,6 @@ setuptools.setup(
     scripts=['tools/cloud-init-per'],
     license='GPLv3',
     data_files=data_files,
-    install_requires=requirements,
     cmdclass=cmdclass,
     entry_points={
         'console_scripts': [
Index: cloud-init-0.7.8/cloudinit/settings.py
===================================================================
--- cloud-init-0.7.8.orig/cloudinit/settings.py
+++ cloud-init-0.7.8/cloudinit/settings.py
@@ -48,13 +48,16 @@ CFG_BUILTIN = {
     ],
     'def_log_file': '/var/log/cloud-init.log',
     'log_cfgs': [],
-    'syslog_fix_perms': ['syslog:adm', 'root:adm'],
+    'mount_default_fields': [None, None, 'auto', 'defaults,nofail', '0', '2'],
+    'ssh_deletekeys': False,
+    'ssh_genkeytypes': [],
+    'syslog_fix_perms': [],
     'system_info': {
         'paths': {
             'cloud_dir': '/var/lib/cloud',
             'templates_dir': '/etc/cloud/templates/',
         },
-        'distro': 'ubuntu',
+        'distro': 'fedora',
     },
     'vendor_data': {'enabled': True, 'prefix': []},
 }
