Index: cloud-init-0.7.6-bzr1245/cloudinit/ssh_util.py
===================================================================
--- cloud-init-0.7.6-bzr1245.orig/cloudinit/ssh_util.py
+++ cloud-init-0.7.6-bzr1245/cloudinit/ssh_util.py
@@ -33,6 +33,7 @@ DEF_SSHD_CFG = "/etc/ssh/sshd_config"
 # taken from openssh source key.c/key_type_from_name
 VALID_KEY_TYPES = (
     "rsa", "dsa", "ssh-rsa", "ssh-dss", "ecdsa",
+    "ecdsa-sha2-nistp256", "ecdsa-sha2-nistp384", "ecdsa-sha2-nistp521",
     "ssh-rsa-cert-v00@openssh.com", "ssh-dss-cert-v00@openssh.com",
     "ssh-rsa-cert-v00@openssh.com", "ssh-dss-cert-v00@openssh.com",
     "ssh-rsa-cert-v01@openssh.com", "ssh-dss-cert-v01@openssh.com",
